### PR TITLE
Add generic operator install role

### PR DESCRIPTION
**Pull request by wkulhanek**
##### SUMMARY

Added a generic role, *install_operator* that installs an Operator into an OpenShift Cluster. It supports all the usual features like manual approval or using a catalog source.

Also added a workload, *ocp4_workload_openshift_gitops* using that generic role as a first example. Workloads will be updated to use the generic role as they are being updated periodically.

@thoraxe FYI.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
install_operator
ocp4_workload_openshift_gitops

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by rut31337 — APPROVED** [review]

File: ansible/roles/install_operator/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ocp_username: opentlc-mgr
+silent: false
+
+# Install or remove the operator
+install_operator_action: install
+# install_operator_action: remove
+
+# Name of the Operator. Must match the name of the PackageManifest
+# in the openshift-marketplace project (or wherever the custom catalogsource is installed)
+install_operator_name: openshift-pipelines
```
[the remark this answers is not in the document]
Comment (wkulhanek) [reply ansible/roles/install_operator/defaults/main.yml:12]: Done. And added assert to test for it.

File: ansible/roles/install_operator/tasks/install.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
+---
+- name: Create Catalogsource for use with catalog snapshot
+  when: install_operator_use_catalog_snapshot | bool
+  k8s:
+    state: present
+    definition: "{{ lookup('template', './templates/catalogsource.j2' ) | from_yaml }}"
```
[the remark this answers is not in the document]
Comment (wkulhanek) [reply ansible/roles/install_operator/tasks/install.yml:6]: Done.

File: ansible/roles/install_operator/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+---
+# Do not modify this file
+
+- name: Install the operator
+  when: install_operator_action == "install"
+  include_tasks:
+    file: ./install.yml
```
[the remark this answers is not in the document]
Comment (wkulhanek) [reply ansible/roles/install_operator/tasks/main.yml:7]: Done

File: ansible/roles/install_operator/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+- name: Remove the operator
+  when: install_operator_action == "remove"
+  include_tasks:
+    file: ./remove.yml
```
[the remark this answers is not in the document]
Comment (wkulhanek) [reply ansible/roles/install_operator/tasks/main.yml:12]: Done

**Pull request merged**